### PR TITLE
Update global.json to use `6.0.300-preview`

### PR DESCRIPTION
**Pull request by TanayParikh**
In accordance with latest guidance https://github.com/dotnet/maui/blob/main/.github/DEVELOPMENT.md#net-6-sdk.

### CODE REVIEW

File: global.json [hunk 1]
```diff
@@ -1,6 +1,9 @@
 {
+  "sdk": {
+    "version": "6.0.300-preview"
+  },
```
Comment (jonathanpeppers) [inline global.json:4]: Do we actually want this? If you're building MAUI, we should trust you know how to select the .NET you want?

Like I normally change `%PATH%`, depending on what I'm doing.

File: global.json [hunk 2]
```diff
@@ -1,6 +1,9 @@
...
   "tools": {
-    "dotnet": "6.0.100-rc.2.21478.25"
+    "dotnet": "6.0.300-preview"
```
Comment (jonathanpeppers) [inline global.json:6]: This was an entry that dotnet/arcade tooling uses on our internal CI. They have powershell scripts and things that read this number. I think you have to put the full version number to change this, it doesn't know how to roll forward.

Can we just put the latest stable .NET release? Or not change this?

File: global.json (hunk 1 above)
Comment (TanayParikh) [reply global.json:4]: > Like I normally change %PATH%, depending on what I'm doing.

We don't really have any [documentation](https://github.com/dotnet/maui/blob/main/.github/DEVELOPMENT.md#running) surrounding this presently. The case can be made to add documentation to do this (on a per platform level). However, I personally prefer this approach as it enforces you're running on the "correct" dotnet version. And this gets updated locally upon `git pull` as MAUI officially starts supporting new versions of the SDK. Currently, again in my experience, it tends to be a bit hit or miss on whether the newest preview is supported or not. For a while I was using 6.0.300* when we were still technically only supporting 6.0.200*, this lead to a good deal of pain when things randomly didn't work. Having this here also helps serve as a single source of truth of what version of the .NET SDK you _should_ be running against.

File: global.json (hunk 2 above)
Comment (TanayParikh) [reply global.json]: Hmm looks like this is required to get the SDK change to work. 

https://dev.azure.com/xamarin/public/_build/results?buildId=58378&view=logs&jobId=2751d22a-42ce-5ac2-f40c-7f95ad2ff1fb&j=2751d22a-42ce-5ac2-f40c-7f95ad2ff1fb&t=70ac8f3a-d989-5549-c519-2cd4c5147626

File: global.json [hunk 3]
```diff
@@ -1,6 +1,9 @@
 {
+  "sdk": {
+    "version": "6.0.300-preview"
```
Comment (jonathanpeppers) [inline global.json:3]: Could we also add:
```suggestion
    "version": "6.0.300-preview",
    "rollForward": "latestMinor",
    "allowPrerelease": true
```

**Comment by rmarinho** [comment]
this is in place just for some arcade things to work irc. it's not related with MAUI version it's using per se.

File: global.json (hunk 2 above)
Comment (rmarinho) [reply global.json]: Nop, we should use the latest stable. this is only used like @jonathanpeppers said for CI. I can also try to remove this need if that's a issue. Should we fix the script @jonathanpeppers  ? maybe hardcode this value on the script?

**Comment by jonathanpeppers** [comment]
@rmarinho it's something from arcade, so I don't actually know how it works. If dotnet/maui was fully on arcade, then I think Maestro can handle updating the number.

Might be worth just putting the latest stable .NET on there for now.

File: global.json [hunk 4]
```diff
@@ -1,6 +1,11 @@
 {
+  "sdk": {
+    "version": "6.0.300",
+    "rollForward": "latestMinor",
+    "allowPrerelease": true
+  },
   "tools": {
-    "dotnet": "6.0.100-rc.2.21478.25"
+    "dotnet": "6.0.300-preview.22173.2"
```
Comment (jonathanpeppers) [inline global.json:8]: Error is:
```
      A compatible installed .NET SDK for global.json version [6.0.300] from [/Users/builder/azdo/_work/1/s/global.json] was not found.
      Install the [6.0.300] .NET SDK or update [/Users/builder/azdo/_work/1/s/global.json] with an installed .NET SDK:
        6.0.201 [/Users/builder/azdo/_work/_tool/dotnet/sdk]
```

Maybe try latest stable?
```suggestion
    "dotnet": "6.0.201"
```

File: global.json [hunk 5]
```diff
@@ -1,6 +1,11 @@
 {
+  "sdk": {
+    "version": "6.0.300",
```
Comment (jonathanpeppers) [inline global.json:3]: I just installed .NET 7 to work on something, and I found I needed this to build `dotnet/maui` afterward:
```suggestion
    "version": "6.0.300-preview",
```
I'm not sure why it wasn't rolling forward? Maybe previews are different?

**Review by Redth — APPROVED** [review]

**Pull request merged**